stmtcacher.go

```diff
@@ -66,6 +81,26 @@ func (sc *stmtCacher) QueryRow(query string, args ...interface{}) RowScanner {
 	return stmt.QueryRow(args...)
 }
 
+func (sc *stmtCacher) Clear() error {
+	sc.mu.Lock()
+	defer sc.mu.Unlock()
+
+	for key, stmt := range sc.cache {
+		delete(sc.cache, key)
+
+		if stmt == nil {
+			continue
+		}
+
+		if err := stmt.Close(); err != nil {
+			return err
```

Comment:
I think this should not short-circuit but instead collapse errors in some way. In this context its probably acceptable to just e.g. `Errorf("one or more Stmt.Close failed; last error: %v", err)`